Enable uploading RBE test results to BQ

matt-kwong opened the pull request.

grpc-testing commented:
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

grpc-testing commented:
```
[trickle] No significant performance differences
```

grpc-testing commented:
```
[microbenchmarks] No significant performance differences
```

grpc-testing commented:
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

grpc-testing commented:
```
[trickle] No significant performance differences
```

grpc-testing commented:
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

grpc-testing commented:
```
[trickle] No significant performance differences
```

grpc-testing commented:
```
[microbenchmarks] No significant performance differences
```

matt-kwong commented:
I rebased the PR to fix merge conflicts and removed `export UPLOAD_TEST_RESULTS=true` from the pull request scripts. Sample results from this RBE builds on this PR are in BQ.

jtattermusch commented on tools/internal_ci/linux/grpc_bazel_on_foundry_base.sh:
```diff
@@ -57,6 +57,8 @@ source tools/internal_ci/helper_scripts/prepare_build_linux_rc
 
 if [ "$UPLOAD_TEST_RESULTS" != "" ]
 then
+  # Sleep to let ResultStore finish writing results before querying
```
is there a risk of writing partial test results if result store takes longer or is this binary (result present / results not present -> error)?

matt-kwong replied:
I think it's binary. IIRC, the error stacktrace originates from the first call to ResultStore's API, so I assume it didn't partially upload, but I didn't check. 

I don't think we're at risk of having results being partially uploaded without the entire build failing.

grpc-testing commented:
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

jtattermusch commented on tools/internal_ci/linux/pull_request/grpc_bazel_on_foundry_dbg.sh:
```diff
@@ -15,4 +15,3 @@
 
 EXTRA_FLAGS="-c dbg --test_timeout=300,450,1200,3600"
```
`export UPLOAD_TEST_RESULTS=true` missing here?

matt-kwong replied:
I don't think we want to collect results from pull request runs since it won't help in determining how flaky a test target is.

jtattermusch replied:
ah, yes, you're right I didn't see the `pull_request` part.

jtattermusch commented on tools/internal_ci/linux/pull_request/grpc_bazel_on_foundry_opt.sh:
```diff
@@ -13,5 +13,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+export UPLOAD_TEST_RESULTS=true
```
This is on pull requests too, let's remove the `export UPLOAD_TEST_RESULTS=true`

matt-kwong replied:
Missed this, thanks

grpc-testing commented:
```
[trickle] No significant performance differences
```

matt-kwong commented:
When we get to the state where each test is only invoked once by Bazel or `run_tests.py`, then I agree that it would make sense to keep them on the same table. But there's also the issue of the schemas being different and I don't see an easy way of merging them.

grpc-testing commented:
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

jtattermusch commented:
@matt-kwong  good point on the uniqueness, let's leave as is and revisit in the future (I'm sure there's a way to merge the schemas and the disambiguate which result type is which).

grpc-testing commented:
```
[trickle] No significant performance differences
```

grpc-testing commented:
```
[microbenchmarks] No significant performance differences
```

matt-kwong commented:
Interop failures: #15310

grpc-testing commented:
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

grpc-testing commented:
```
[trickle] No significant performance differences
```

grpc-testing commented:
```
[microbenchmarks] No significant performance differences
```

grpc-testing commented:
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

grpc-testing commented:
```
[trickle] No significant performance differences
```

grpc-testing commented:
```
[microbenchmarks] No significant performance differences
```
